reactive/kotlinx-coroutines-reactor/src/Mono.kt

```diff
@@ -34,6 +32,17 @@ public fun <T> mono(
     return monoInternal(GlobalScope, context, block)
 }
 
+/**
+ * Awaits the completion of the [Mono] without blocking the thread.
+ * Returns the resulting value, or `null` if no value is produced, or throws the corresponding exception if this
+ * [Mono] has produced an error.
+ *
+ * This suspending function is cancellable.
+ * If the [Job] of the current coroutine is cancelled or completed while this suspending function is waiting, this
+ * function immediately resumes with [CancellationException] and cancels its subscription.
+ */
+public suspend fun <T> Mono<T>.await(): T? = awaitFirstOrNull()
```

Comment:
It seems like the `awaitOrNull` can be a more appropriate name.
We can discuss it along with `singleOr*` deprecation if you don't mind

Reply:
Or, alternatively, `awaitCompletion`?

Reply:
Wouldn't we have to rename the existing `MaybeSource<T>.await()` then?

Reply:
Good point, I'll revisit them